### ISSUE TITLE
Support custom tutorial per project

### ISSUE THREAD

**Issue by laurentS** [mapswipe/mapswipe#190]
The tutorial content is currently defined by the project type. We should make it possible for project managers to select a more specific tutorial when creating a project, so that instructions, and particularly imagery can be tailored to the environments seen in various countries.

If no tutorial is specified, it should fallback to the current default value.

**Comment by cgiovando** [comment]
Hi @laurentS - thanks for tracking this. I'd be happy to test the creation of custom tutorials when the functionality is ready. I remember @Hagellach37 mentioning a JSON template file, where tutorial parameters and images can be customized for each project? I tried to dig through the code, but could not find it. Any pointer would be appreciated - thanks!

**Comment by laurentS** [comment]
@cgiovando You can already test out the custom tutorials in `2.0.2(6)-beta` on the app side, however you'd need to create the tutorial by hand.
The tutorial is essentially a mapswipe project with some custom values:
First you need to create the project data: Below is an example for the original CD tutorial. The `pre`, `post_correct` and `post_wrong` keys are the prompts shown before the user answers, and after they check their answer, if they're right or wrong. The names such as `changes_1` match those in the tasks below.
```JSON
{
  "categories" : {
    "changes_1" : {
      "post_correct" : "Yes, that looks like a new settlement being developed. A big new building has been built here. Swipe left to continue the tutorial.",
      "post_wrong" : "Oh no. Make sure to spot all the new settlements and swipe right. New roads can also function as an indicator sometimes.",
      "pre" : "For this project we are looking for changes regarding settlements. Compare the upper (before) and lower (after) image and *SWIPE RIGHT* (not tap) to indicate that new settlements have been built."
    },
    "changes_2" : {
      "post_correct" : "Correct. We can spot new settlements in the lower left part of the satellite image. Swipe left to go on with some more examples.",
      "post_wrong" : "Take a closer look. There are new settlements here in the lower left part of the satellite image.",
      "pre" : "And again: If new settlements have been built, *SWIPE* to the right side."
    },
    "changes_3" : {
      "post_correct" : "Correct. We can detect some settlements that have been removed here.",
      "post_wrong" : "Take a closer look. There are settlements in the upper image that are not there in lower one.",
      "pre" : "Sometimes settlement might also disappear because they have been pulled down or destroyed. Also for this case *SWIPE RIGHT* to indicate that the situation changed."
    },
    "maybe_1" : {
      "post_correct" : "Great. We cannot really tell if the building at the bottom left is a new settlement. This is a 'maybe'.",
      "post_wrong" : "Are you really sure? Sometimes it's better to mark the imagery as 'not sure'.",
      "pre" : "Sometimes it might be unclear if settlements emerged. Swipe down if you are not sure."
    },
    "no_changes_1" : {
      "post_correct" : "Cool. You got it. Just look out for changes concerning settlements. Swipe left to continue.",
      "post_wrong" : "Oh no. Not all changes refer to new settlements, some are just related to agriculture or vegetation.",
      "pre" : "In this example, we might spot changes in the vegetation for this imagery pair, but no changes related to settlements. *SWIPE* left to indicate no changes."
    },
    "no_changes_2" : {
      "post_correct" : "Nice. Just watch out for changes referring to settlements.",
      "post_wrong" : "Sorry, try again. Don't be distracted by changes in vegetation, just look out for settlements.",
      "pre" : "There is a building but *NO CHANGES* in settlements can be observed here. Again, *SWIPE LEFT*."
    },
    "no_changes_3" : {
      "post_correct" : "Nice. Don't be distracted by the shift in the satellite image.",
      "post_wrong" : "Sorry, try again. Don't be distracted by the shift in the satellite image.",
      "pre" : "In this example there are already many buildings. But we can't see changes, just a little shift. *SWIPE LEFT*."
    }
  },
  "contributorCount" : 0,
  "examplesFile" : "sample_data/change_detection_tutorial_tasks_kutupalong.geojson",
  "lookFor" : "CHANGES IN BUILDINGS",
  "name" : "Change Detection Tutorial",
  "progress" : 0,
  "projectDetails" : "This is a tutorial",
  "projectId" : "change_detection_tutorial",
  "projectType" : 3,
  "status" : "change_detection_tutorial",
  "tileServerA" : {
    "apiKey" : "AopsdXjtTu-IwNoCTiZBtgRJ1g7yPkzAi65nXplc-eLJwZHYlAIf2yuSY_Kjg3Wn",
    "captions" : "",
    "credits" : "International Organization for Migration (IOM)",
    "date" : "",
    "name" : "custom",
    "url" : "https://tiles.openaerialmap.org/5a4203c12553e6000ce5ab28/0/b6dae42c-0c6e-4f8e-ac84-a1344aa009da/{z}/{x}/{y}.png"
  },
  "tileServerB" : {
    "apiKey" : "91e57457-aa2d-41ad-a42b-3b63a123f54a",
    "captions" : "",
    "credits" : "International Organization for Migration (IOM)",
    "date" : "",
    "name" : "custom",
    "url" : "https://tiles.openaerialmap.org/5ae862e00b093000130affd9/0/5ae862e00b093000130affda/{z}/{x}/{y}.png"
  },
  "zoomLevel" : 18
}
```
Then you need a group JSON, like this one:
```JSON
{
  "finishedCount" : 0,
  "groupId" : 101,
  "numberOfTasks" : 4,
  "progress" : 0,
  "projectId" : "change_detection_tutorial",
  "requiredCount" : 5,
  "xMax" : "105",
  "xMin" : "100",
  "yMax" : "32768",
  "yMin" : "32768"
}
```
And finally a few tasks. This defines 6 screens, I've added the explanation for each key after them below (so the JSON is not valid for the first screen):
```JSON
[ {
  "category" : "changes_1",   // matches the strings in the project data above
  "groupId" : 101,   // matches the groupId above
  "projectId" : "change_detection_tutorial",   // the name of the tutorial, must match between the various bits
  "referenceAnswer" : 1,   // what answer is expect (0 to 3, which matches then number of taps)
  "taskId" : "16-100-32768",  // zoom-taskX-taskY for this tutorial
  "taskId_real" : "20-792684-461213",  // the original imagery coordinates
  "taskX" : 100,
  "taskY" : 32768,
  "urlA" : "https://tiles.openaerialmap.org/5a4203c12553e6000ce5ab28/0/b6dae42c-0c6e-4f8e-ac84-a1344aa009da/20/792684/461213.png",   // url for the top image...
  "urlB" : "https://tiles.openaerialmap.org/5ae862e00b093000130affd9/0/5ae862e00b093000130affda/20/792684/461213.png"
}, {
  "category" : "changes_2",
  "groupId" : 101,
  "projectId" : "change_detection_tutorial",
  "referenceAnswer" : 1,
  "taskId" : "16-101-32768",
  "taskId_real" : "20-792667-461206",
  "taskX" : 101,
  "taskY" : 32768,
  "urlA" : "https://tiles.openaerialmap.org/5a4203c12553e6000ce5ab28/0/b6dae42c-0c6e-4f8e-ac84-a1344aa009da/20/792667/461206.png",
  "urlB" : "https://tiles.openaerialmap.org/5ae862e00b093000130affd9/0/5ae862e00b093000130affda/20/792667/461206.png"
}, {
  "category" : "changes_3",
  "groupId" : 101,
  "projectId" : "change_detection_tutorial",
  "referenceAnswer" : 1,
  "taskId" : "16-102-32768",
  "taskId_real" : "20-792697-461224",
  "taskX" : 102,
  "taskY" : 32768,
  "urlA" : "https://tiles.openaerialmap.org/5a4203c12553e6000ce5ab28/0/b6dae42c-0c6e-4f8e-ac84-a1344aa009da/20/792697/461224.png",
  "urlB" : "https://tiles.openaerialmap.org/5ae862e00b093000130affd9/0/5ae862e00b093000130affda/20/792697/461224.png"
}, {
  "category" : "no_changes_1",
  "groupId" : 101,
  "projectId" : "change_detection_tutorial",
  "referenceAnswer" : 0,
  "taskId" : "16-103-32768",
  "taskId_real" : "20-792655-461208",
  "taskX" : 103,
  "taskY" : 32768,
  "urlA" : "https://tiles.openaerialmap.org/5a4203c12553e6000ce5ab28/0/b6dae42c-0c6e-4f8e-ac84-a1344aa009da/20/792655/461208.png",
  "urlB" : "https://tiles.openaerialmap.org/5ae862e00b093000130affd9/0/5ae862e00b093000130affda/20/792655/461208.png"
}, {
  "category" : "no_changes_2",
  "groupId" : 101,
  "projectId" : "change_detection_tutorial",
  "referenceAnswer" : 0,
  "taskId" : "16-104-32768",
  "taskId_real" : "20-792651-461208",
  "taskX" : 104,
  "taskY" : 32768,
  "urlA" : "https://tiles.openaerialmap.org/5a4203c12553e6000ce5ab28/0/b6dae42c-0c6e-4f8e-ac84-a1344aa009da/20/792651/461208.png",
  "urlB" : "https://tiles.openaerialmap.org/5ae862e00b093000130affd9/0/5ae862e00b093000130affda/20/792651/461208.png"
}, {
  "category" : "no_changes_3",
  "groupId" : 101,
  "projectId" : "change_detection_tutorial",
  "referenceAnswer" : 0,
  "taskId" : "16-105-32768",
  "taskId_real" : "20-792685-461212",
  "taskX" : 105,
  "taskY" : 32768,
  "urlA" : "https://tiles.openaerialmap.org/5a4203c12553e6000ce5ab28/0/b6dae42c-0c6e-4f8e-ac84-a1344aa009da/20/792685/461212.png",
  "urlB" : "https://tiles.openaerialmap.org/5ae862e00b093000130affd9/0/5ae862e00b093000130affda/20/792685/461212.png"
}, {
  "category" : "maybe_1",
  "groupId" : 101,
  "projectId" : "change_detection_tutorial",
  "referenceAnswer" : 2,
  "taskId" : "16-106-32768",
  "taskId_real" : "20-792661-461202",
  "taskX" : 106,
  "taskY" : 32768,
  "urlA" : "https://tiles.openaerialmap.org/5a4203c12553e6000ce5ab28/0/b6dae42c-0c6e-4f8e-ac84-a1344aa009da/20/792661/461202.png",
  "urlB" : "https://tiles.openaerialmap.org/5ae862e00b093000130affd9/0/5ae862e00b093000130affda/20/792661/461202.png"
} ]
```

If you copy/paste these and adjust the content to what you want, we should be able to inject them back into the firebase tree and try it out. Let me know if that helps!

**Comment by laurentS** [comment]
Released in 2.0.3
